hollow-diff-ui/src/main/java/com/netflix/hollow/history/ui/HollowHistoryUI.java

```diff
@@ -77,16 +77,53 @@ public HollowHistoryUI(String baseUrlPath, HollowConsumer consumer, int numState
     }
     
     private static HollowHistory createHistory(HollowConsumer consumer, int numStatesToTrack) {
-        consumer.getRefreshLock().lock();
-        try {
-            HollowHistory history = new HollowHistory(consumer.getStateEngine(), consumer.getCurrentVersionId(), numStatesToTrack);
-            consumer.addRefreshListener(new HollowHistoryRefreshListener(history));
-            return history;
-        } finally {
-            consumer.getRefreshLock().unlock();
+        return createHistory(consumer, null, numStatesToTrack);
+    }
+
+    private static HollowHistory createHistory(HollowConsumer consumerFwd, HollowConsumer consumerRev, int numStatesToTrack) {
```

Comment:
The underlying `HollowHistory` initialization has a 
```
requireNonNull(fwdMovingHollowReadStateEngine, "Fwd direction read state engine should be initialized");
```
and yes, we don't support rev only because it isn't really useful and it would add complexity- api that accepts a single state engine would need to also accept/determine which direction it will be traversing in